codebasin/report.py

```diff
@@ -244,3 +254,454 @@ def clustering(output_name, setmap):
         fig.savefig(fp)
 
     return "\n".join(lines)
+
+
+def _human_readable(x: int) -> str:
+    """
+    Parameters
+    ----------
+    x: int
+        An integer.
+
+    Returns
+    -------
+    str
+        A human-readable representation of x.
+
+    Raises
+    ------
+    TypeError
+        If `x` is not an integer.
+    """
+    if not isinstance(x, numbers.Integral):
+        raise TypeError("x must be Integral")
+
+    digits = len(str(x))
+    if digits <= 3:
+        return str(x)
+    elif digits <= 6:
+        return f"{x/10**3:.1f}k"
+    elif digits <= 9:
+        return f"{x/10**6:.1f}M"
+    elif digits <= 12:
+        return f"{x/10**9:.1f}G"
+    return "??????"
```

Comment:
Maybe `*******` instead?

Reply:
Fixed in 987692c139dd36a68f44a3a07466e889a4b014e5.